sycl/include/CL/sycl/types.hpp

```diff
@@ -644,30 +644,38 @@ template <typename Type, int NumElements> class vec {
 #undef __SYCL_ACCESS_RETURN
   // End of hi/lo, even/odd, xyzw, and rgba swizzles.
 
-  // TODO: make templated address space to work.
-  // Somehow, access<> to multi_ptr<> conversion doesn't work w/o making
-  // address space explicitly specified.
-#ifdef __SYCL_LOADSTORE
-#error "Undefine __SYCL_LOADSTORE macro"
-#endif
-#define __SYCL_LOADSTORE(Space)                                                \
-  void load(size_t Offset, multi_ptr<const DataT, Space> Ptr) {                \
-    for (int I = 0; I < NumElements; I++) {                                    \
-      setValue(                                                                \
-          I, *multi_ptr<const DataT, Space>(Ptr + Offset * NumElements + I));  \
-    }                                                                          \
-  }                                                                            \
-  void store(size_t Offset, multi_ptr<DataT, Space> Ptr) const {               \
-    for (int I = 0; I < NumElements; I++) {                                    \
-      *multi_ptr<DataT, Space>(Ptr + Offset * NumElements + I) = getValue(I);  \
-    }                                                                          \
+  template <access::address_space Space>
+  void load(size_t Offset, multi_ptr<const DataT, Space> Ptr) {
+    for (int I = 0; I < NumElements; I++) {
+      setValue(I,
+               *multi_ptr<const DataT, Space>(Ptr + Offset * NumElements + I));
+    }
+  }
+  template <access::address_space Space>
+  void load(size_t Offset, multi_ptr<DataT, Space> Ptr) {
+    multi_ptr<const DataT, Space> ConstPtr(Ptr);
+    load(Offset, ConstPtr);
+  }
+  template <int Dimensions, access::mode Mode,
+            access::placeholder IsPlaceholder, access::target Target>
+  void load(size_t Offset,
+            accessor<DataT, Dimensions, Mode, Target, IsPlaceholder> Acc) {
+    multi_ptr<const DataT, detail::TargetToAS<Target>::AS> MultiPtr(Acc);
+    load(Offset, MultiPtr);
+  }
```

Comment:
Cannot find such overload in the spec

Reply:
@AlexeySachkov but it is being used in CTS

Reply:
@alexbatashev, then it is either CTS bug or spec bug.

Tagging @bader, @mkinsner and @jbrodman for visibility

Reply:
@AlexeySachkov I submitted an issue to track this question https://github.com/KhronosGroup/SYCL-CTS/issues/36